### DISCUSSION TITLE
Samsung 2022 protocol

### DISCUSSION THREAD

**Discussion by societyofrobots** [ukw100/IRMP#83] [Q&A]
I noticed that the protocol for Samsung TV's stop working starting with 2022 models. They are using some new proprietary protocol. None of the protocols in the extended IRMP list is working for it.

I'm hoping/wondering if someone here already has code for this?

**Comment by eku** [comment]
You may get support in the forum https://www.mikrocontroller.net/topic/irmp-infrared-multi-protocol-decoder

**Comment by societyofrobots** [comment]
I tried doing the logging method in your link, but ran into [this issue / bug?](https://github.com/IRMP-org/IRMP/issues/84).

So, I added this in my IR loop:
`Serial.print(irmp_data.command);`

I push Power on each of three remotes. They all return SAMSG32, but the commands are (annoyingly) different for each.

2020 Samsung remote gives me:
```
1P=SAMSG32 | 0xA A=0x707 C=0x19E6
6630, P=SAMSG32 | 0xA A=0x707 C=0x19E6 R
6630, P=SAMSG32 | 0xA A=0x707 C=0x19E6 R
64770
```

2022 Samsung remote gives me:
```
P=SAMSG32 | 0xA A=0x707 C=0xFD02
64770, P=SAMSG32 | 0xA A=0x707 C=0xFD02 R
11985, P=SAMSG32 | 0xA A=0x707 C=0x2ED1
11985, P=SAMSG32 | 0xA A=0x707 C=0x2ED1 R
11985, P=SAMSG32 | 0xA A=0x707 C=0x2ED1 R
11985, P=SAMSG32 | 0xA A=0x707 C=0x2ED1 R
```

Universal remote gives me:
```
P=SAMSG32 | 0xA A=0x707 C=0xFD02
64770, P=SAMSG32 | 0xA A=0x707 C=0xFD02 R
```

I can then use C as the command to look for.

However, the new 2022 remote is returning C=0x2ED1 regardless of what button I push on the remote. Ideas?

edit: I just found out that when fresh out of the box or factory reset, every remote button will send the same IR code, which initiates BT pairing. That's probably the 2ED1. I'll update this the more I figure out.
